tools/sz_repo_cli/lib/src/commands/src/deploy_ios_command.dart

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2022 Sharezone UG (haftungsbeschränkt)
+// Licensed under the EUPL-1.2-or-later.
+//
+// You may obtain a copy of the Licence at:
+// https://joinup.ec.europa.eu/software/page/eupl
+//
+// SPDX-License-Identifier: EUPL-1.2
+
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:sz_repo_cli/src/common/common.dart';
+
+final _iosStages = [
+  'stable',
+  'alpha',
+];
+
+/// The different flavors of the iOS app that support deployment.
+final _iosFlavors = [
+  'prod',
+];
+
+/// [DeployIosCommand] provides functionality for deploying the Sharezone iOS
+/// app to the App Store or TestFlight.
+///
+/// This command automatically increments the build number and builds the app.
+/// The Codemagic CLI tools are required for this process. Note that only the
+/// "prod" flavor of the app is currently supported for iOS deployment.
+///
+/// You can customize deployment using command line arguments. Some of these
+/// include:
+///  - `private-key`: The App Store Connect API private key used for JWT
+///    authentication.
+///  - `key-id`: The App Store Connect API Key ID used to authenticate.
+///  - `issuer-id`: The App Store Connect API Key Issuer ID used to
+///    authenticate.
+///  - `stage`: The stage to deploy to. Supports "stable" for App Store releases
+///    and "alpha" for TestFlight releases.
+///  - `flavor`: The flavor to build for. Currently only "prod" flavor is
+///    supported.
+///  - `whats-new`: Release notes either for TestFlight or App Store review
+///    submission.
+///
+/// These options can either be provided via the command line or set as
+/// environment variables (only applies for some of them). If any required
+/// argument is missing, the deployment will fail.
+class DeployIosCommand extends Command {
+  final SharezoneRepo _repo;
+
+  DeployIosCommand(this._repo) {
+    argParser
+      ..addOption(
+        releaseStageOptionName,
+        abbr: 's',
+        allowed: _iosStages,
+        help:
+            'The stage to deploy to. The "stable" stage is used for App Store '
+            'releases, the "alpha" stage is used for TestFlight releases.',
+        defaultsTo: 'stable',
+      )
```

Comment:
> Random note: For the deploy web command the have the stages [alpha, beta, prod] instead of [alpha, beta, stable]. I guess these should be the same everywhere?

For now, we don't have an iOS beta track. Web is the only platform with a beta track (which btw we don't use xD).

> Also can you describe in the help what the stages do exactly and how they are used? From what I gathered "--stage alpha" will need a matching "beta group" in TestFlight called "alpha". Additonally the stage is used in the build process, what will it change there? Is there a change?

Do you think it's a good idea to describe what the stage parameter changes in the build process / in the app at the help string of the iOS deploy command? I'll also create a PR for `sz deploy android` and `sz deploy macos` command. Changing the behavior how `--dart-define=DEVELOPEMTENT_STAGE` affects the app behavior would require to update the documentation at 5 difference locations (inside the app, at `sz deploy ios`, `sz deploy android`, `sz deploy web` and `sz deploy macos`).

My current draft of the `help` text is:

The stage to deploy to. The "stable" stage is used for App Store releases, the "alpha" stage is used for TestFlight releases (publishing to a TestFligh group that matches "alpha" as a group name). Additionally, the stage value is passed to the build command.

Reply:
It was just confusing for me because for me just reading the argument description I thought it was **only** for TestFlight which confused me because I saw it was also passed to the build command (I forgot all about stages :D).

So I guess one thing I don't like is that the stage name and the TestFlight beta group names are coupled. In my opinion they should be decoupled so that we could change the stage names and not break anything for TestFlight. Decoupling it would have also helped me to understand that the stage string is not primarily for TestFlight, but our definition of deployment stages. They could be decoupled like in the `deploy web-app` code (just a simple map should be enough I guess).

> Do you think it's a good idea to describe what the stage parameter changes in the build process / in the app at the help string of the iOS deploy command? I'll also create a PR for sz deploy android and sz deploy macos command. Changing the behavior how --dart-define=DEVELOPEMTENT_STAGE affects the app behavior would require to update the documentation at 5 difference locations (inside the app, at sz deploy ios, sz deploy android, sz deploy web and sz deploy macos).

Yeah you're right, a detailed description would be bad. 

I like your description. If we decoupled the TestFlight stage name then we could leave the stuff about the beta group out in my opinion.

What about this?
> The deployment stage to deploy to. The "stable" stage is used for App Store releases, the "alpha" stage is used for TestFlight releases. The value will be forwarded to the "sz build" command (if a fresh build is necessary).

Reply:
I'm using now the following the `help` value for `--stage`:

"The deployment stage to deploy to. The "stable" stage is used for App Store releases, the "alpha" stage is used for TestFlight releases. The value will be forwarded to the "sz build" command "

I also decoupled the stage and the TestFligh Group names 👍